ManifestData.cs

```diff
@@ -23,251 +31,332 @@ public class ManifestData {
-      data[243520][322022] = new Datum(1221230661228297401, 4, 291798721);
-      data[226087][322022] = new Datum(7180495594989737482, 4, 275966134);
-      data[224995][322022] = new Datum(6536174967355028889, 3, 243487857);
-      data[224531][322022] = new Datum(7516537008051594086, 3, 243488546);
-      data[223249][322022] = new Datum(3867407920981508845, 3, 225475676);
-      data[222477][322022] = new Datum(8917339509592123029, 3, 225475410);
-      data[221394][322022] = new Datum(6888774931510513164, 3, 218106434);
-      data[220996][322022] = new Datum(7673940384630485679, 3, 211726262);
-      data[220675][322022] = new Datum(4383443687812477256, 3, 206909511);
-      data[220625][322022] = new Datum(8183524469048336858, 3, 206907989);
-      data[220480][322022] = new Datum(2824340380872620137, 3, 172228256);
+      return totalDownloadSize;
+    }
+
+    public static IEnumerable<SteamManifest> GetManifestsForVersion(int version, Package package = Package.All) {
+      return data[version].Where(m => package == Package.All || package == m.package);
+    }
+
+    private static readonly Dictionary<int, List<SteamManifest>> data = InitData();
+    private static Dictionary<int, List<SteamManifest>> InitData() {
+      Dictionary<int, List<SteamManifest>> data = new Dictionary<int, List<SteamManifest>>();
+
+      data[220480] = new List<SteamManifest>() {
+        new SteamManifest(257510, 257511, 4122334240058475272, 2, 48908496, Package.Main),
+        new SteamManifest(257510, 257515, 1348826399794024471, 32, 5055625401, Package.Main),
+        new SteamManifest(257510, 257516, 1237958166729860756, 6, 117245, Package.Main),
+        new SteamManifest(257510, 257519, 2322040020544521685, 1, 1739595, Package.Main),
+        new SteamManifest(257510, 322022, 2824340380872620137, 3, 172228256, Package.Prototype)
+      };
+
+      data[220625] = new List<SteamManifest> {
+        new SteamManifest(257510, 257511, 8087374737096894864,  2,   48901328, Package.Main),
+        new SteamManifest(257510, 257515, 3617089317964903627, 32, 5217247608, Package.Main),
+        new SteamManifest(257510, 257516, 1237958166729860756,  6,     117245, Package.Main),
+        new SteamManifest(257510, 257519, 1272083368566319066,  1,    1789447, Package.Main),
+        new SteamManifest(257510, 322022, 8183524469048336858,  3,  206907989, Package.Prototype)
+      };
+
+      data[220675] = new List<SteamManifest> {
+        new SteamManifest(257510, 257511, 2018192542376932921,  2,   48900816, Package.Main),
+        new SteamManifest(257510, 257515, 8986158524600611840, 32, 5217247413, Package.Main),
+        new SteamManifest(257510, 257516, 1237958166729860756,  6,     117245, Package.Main),
+        new SteamManifest(257510, 257519, 4554990112195176406,  1,    1789447, Package.Main),
+        new SteamManifest(257510, 322022, 4383443687812477256,  3,  206909511, Package.Prototype)
+      };
+
+      data[220996] = new List<SteamManifest> {

```

Comment:
This new ordering is less helpful, since adding new packages is now an interleaved process, rather than a cumulative one. It also separates similarities, which in my experience makes copy/paste errors harder to spot. Is there any particular reason you felt the need to change this?

Reply:
With the old ordering adding a new version was an interleaved process instead. There isn't really a way to avoid it for both, one of the two has to be interleaved like that. Adding new versions is almost certainly going to be more common than packages, so I switched it.

Somewhat relatedly: I feel like it might be better to load all this data from a file, it becomes easier to update and it should be possible to check a url to add any new versions without updating the whole application. That goes beyond the scope of this PR though.

Reply:
There's only 5-10 packages, so adding a new version is only in 5-10 places.
However, you're planning on adding in editor packages, and there are >30 versions.

I think the long-term solution is storing it in a table so that we can easily re-sort the data. I think WPF has some sort of "data" format?

I don't know why you think a URL is necessary, given how infrequently this game is being updated. However, if you want to move this data table into another file, that is fine.

Reply:
Adding editor is a one time cost, as far as we know they'll keep updating the game indefinitely and we'll have to keep adding new versions, which is certainly much easier when everything is in one place.

Any sort of re-sortable data format also won't fix the actual "issue" here, the code layout. Looking at parsing it from a file though, I do think the same version based format would be better, it mirrors the same sort of structure as the internal dictionary - `data[<version>][<package>][<index>].attribute`.

For now I'll just commit the other fixes and leave this alone, but if it's really such an issue I'll change it in the next one, not like it's particularly hard to rearrange this stuff.